website/content/posts/evaluating-custom-predicates.md

```diff
@@ -0,0 +1,196 @@
+the most popular format is [SLSA Provenance](https://slsa.dev/spec/v0.1/provenance#schema) which is
+used to provide information on how the image created. Our [Hitchhiker’s
+Guide](https://enterprisecontract.dev/docs/user-guide/main/hitchhikers-guide.html) demonstrates how
+to write policies to assert the contents of the SLSA Provenance. Here, we expand on that approach to
+assert the contents of *any* attestation format, even completely made up ones.
+
+<!--more-->
+
+Before getting started, let's make sure we have an image that is already signed and has a SLSA
+Provenance attestation. We will also need access to the signing key used. The [Hitchhiker’s
+Guide](https://enterprisecontract.dev/docs/user-guide/main/hitchhikers-guide.html) walks through the
+process. If you want to try out the commands in this blog post, start there.
+
+When we talk about different attestation formats, what we are really saying is different **predicate
+types** of an in-toto attestation. In addition to SLSA Provenance, there are a few [other common
+ones](https://github.com/in-toto/attestation/blob/main/spec/predicates/README.md). As suggested
+earlier, in this blog post we will use a made up predicate type to both demistify predicates and
+highlight Enterprise Contract's flexibility in validating attestations.
+
+Let's get this started!
+
+First, we create a `spam.json` file with arbitrary JSON data. This holds the contents of our custom
+predicate.
+
+```json
+{"spam": "more spam"}
+```
+
+Second, we create a new attestation for our image using `spam.json` as the predicate for our
+fabricated predicate type.
+
+```bash
+cosign attest --predicate spam.json --type 'https://spam/v42' --key cosign.key $REPOSITORY:latest
+```
+
+Next, we want to write a policy rule that asserts the contents of the spam predicate associated with
+a container image. To do so, it helps to visualize how EC structures this data. The `ec validate
+image` command supports an output format for this specific purpose.
+
+```bash
+ec validate image --public-key cosign.pub --image $REPOSITORY:latest --output yaml \
+  --output policy-input=input.json
+```
+
+Once the command finishes, the `input.json` file contains the complete input as seen by the policy
+rules. Use [jq](https://jqlang.github.io/jq/) to inspect it.
+
+```bash
+$ jq '.attestations[1].statement.predicate' input.json
+{
+    "spam": "more spam"
+}
+```
+
+The "more spam" value can be accessed via the `input.attestations[1].statement.predicate.spam`
+variable from within a policy rule.
+
+Things are getting exciting!
+
+Create a new directory to hold the new policies we are about to write, e.g. `mkdir policies`. Then
+create the file `spam.rego` inside that directory.
+
+```rego
+package spam
+
+import rego.v1
+
+# METADATA
+# title: Spam
+# description: Verify the Spam attestation has the spam attribute
+#   set to the expected value.
+# custom:
+#   short_name: spam
```

Comment:
I was wondering if it would work without the `custom.short_name` since you pass `code` in the result. But I'm guessing it would not show the success result without it - that right?